ompi/datatype/ompi_datatype_module.c

```diff
@@ -547,7 +547,14 @@ int32_t ompi_datatype_init( void )
         ompi_mpi_##name.dt.d_f_to_c_index = index;                      \
         rc = opal_pointer_array_set_item(&ompi_datatype_f_to_c_table,   \
                                          index, &ompi_mpi_##name);      \
-        assert( rc == OPAL_SUCCESS );                                   \
+        /* assert() is compiled out in opt builds, so the compiler      \
+           will throw a ton of warnings about 'rc' not being used.      \
+           Let's just return an error instead of silently failing in    \
+           production.                                                  \
+        */                                                              \
+        if(rc != OPAL_SUCCESS) {                                        \
+            return OMPI_ERROR;                                          \
+        }                                                               \
```

Comment:
assert was for developer's build, nobody should get an error outside us. Returning in a macro is extremely bad practice, way worst that the shadowing.

Reply:
I guess the better way to go here would be to make this a function, because other wise I don't see how we silence the 50 or so warnings rc it produces.

Either that or we just throw the assert away all-together.

Reply:
add `(void)rc;` after the assert ?

Reply:
Sure, that's fine with me.

Reply:
Done.